docs/dev/ffi/when-to-use-what-in-the-ffi.md

```diff
@@ -108,32 +86,32 @@ For return values, used `*mut c_char`, and for input, use
     to the destructor, it will allocate a temporary buffer, pass it to Rust, which
     we'll free, corrupting both heaps 💥. Oops!
 
-2. If the string is passed into Rust from Kotlin/Swift, the rust code should
+2. If the string is passed into Rust from Kotlin/Swift, the Rust code should
    declare the parameter as a [`FfiStr<'_>`](https://docs.rs/ffi-support/*/ffi_support/struct.FfiStr.html).
    and things should then work more or less automatically. The `FfiStr` has methods
    for extracting it's data as `&str`, `Option<&str>`, `String`, and `Option<String>`.
 
-It's also completely fine to use Protobufs or JSON for this case!
-
 ### Aggregates
 
 This is any type that's more complex than a primitive or a string (arrays,
-structures, and combinations there-in). There are two options we recommend for
-these cases:
-
-1. Passing data using Protobufs. See the
-   "[Using Protobufs-encoded data over Rust FFI](https://github.com/mozilla/application-services/blob/master/docs/howtos/passing-protobuf-data-over-ffi.md)"
-   document for details on how to do this. We recommend this for all new use cases, unless
-   you have a specific reason that JSON is better (e.g. semi-opaque JSON encoded data is
-   desired on the other side).
+structures, and combinations there-in).
+There are two options we recommend for these cases:
 
-2. Passing data as JSON. This is very easy, and useful for prototyping, however
+1. Passing data as JSON. This is very easy, and useful for prototyping, however
    much slower, requires a great deal of copying and redundant encode/decode
    steps (in general, the data will be copied at least 4 times to make this
-   work, and almost certainly more in practice), and can be done relatively
-   easily by `derive(Serialize, Deserialize)`, and adding
-   [`ffi_support::implement_into_ffi_by_json`](https://docs.rs/ffi-support/*/ffi_support/macro.implement_into_ffi_by_json.html)
-   into the crate that defines the type.
+   work, and almost certainly more in practice),
+   and can be done relatively easily by `derive(Serialize, Deserialize)`,
+   and converting to a JSON string using `serde_json::to_string`.
```

Comment:
```suggestion
   work, and almost certainly more in practice).
   It can be done relatively easily by `derive(Serialize, Deserialize)`,
   and converting to a JSON string using `serde_json::to_string`.
```